contracts/src/v0.8/llo-feeds/v0.4.0/test/fee-manager/DestinationFeeManager.processFeeBulk.t.sol

```diff
@@ -0,0 +1,256 @@
+    }
+
+    processFee(poolIds, payloads, USER, address(native), DEFAULT_REPORT_NATIVE_FEE * 4);
+
+    assertEq(getNativeBalance(address(feeManager)), DEFAULT_REPORT_NATIVE_FEE * 4);
+    assertEq(getLinkBalance(address(rewardManager)), DEFAULT_REPORT_LINK_FEE * 4);
+    assertEq(getLinkBalance(address(feeManager)), 1);
+
+    assertEq(USER.balance, DEFAULT_NATIVE_MINT_QUANTITY - DEFAULT_REPORT_NATIVE_FEE * 4);
+    assertEq(PROXY.balance, DEFAULT_NATIVE_MINT_QUANTITY);
+  }
+
+  function test_processMultipleV1Reports() public {
+    bytes memory payload = abi.encode(
+      [DEFAULT_CONFIG_DIGEST, 0, 0],
+      getV1Report(DEFAULT_FEED_1_V1),
+      new bytes32[](1),
+      new bytes32[](1),
+      bytes32("")
+    );
+
+    bytes[] memory payloads = new bytes[](NUMBER_OF_REPORTS);
+    for (uint256 i = 0; i < NUMBER_OF_REPORTS; ++i) {
+      payloads[i] = payload;
+    }
+
+    bytes32[] memory poolIds = new bytes32[](NUMBER_OF_REPORTS);
+    for (uint256 i = 0; i < NUMBER_OF_REPORTS; ++i) {
+      poolIds[i] = DEFAULT_CONFIG_DIGEST;
+    }
+
+    processFee(poolIds, payloads, USER, address(native), DEFAULT_REPORT_NATIVE_FEE * 5);
+
+    assertEq(USER.balance, DEFAULT_NATIVE_MINT_QUANTITY);
+    assertEq(PROXY.balance, DEFAULT_NATIVE_MINT_QUANTITY);
+  }
+
+  function test_eventIsEmittedIfNotEnoughLink() public {
+    bytes memory nativePayload = getPayload(getV3Report(DEFAULT_FEED_1_V3));
+
+    bytes[] memory payloads = new bytes[](5);
+    payloads[0] = nativePayload;
+    payloads[1] = nativePayload;
+    payloads[2] = nativePayload;
+    payloads[3] = nativePayload;
+    payloads[4] = nativePayload;
+
+    approveNative(address(feeManager), DEFAULT_REPORT_NATIVE_FEE * 5, USER);
+
+    IDestinationRewardManager.FeePayment[] memory payments = new IDestinationRewardManager.FeePayment[](5);
+    payments[0] = IDestinationRewardManager.FeePayment(DEFAULT_CONFIG_DIGEST, uint192(DEFAULT_REPORT_LINK_FEE));
+    payments[1] = IDestinationRewardManager.FeePayment(DEFAULT_CONFIG_DIGEST, uint192(DEFAULT_REPORT_LINK_FEE));
+    payments[2] = IDestinationRewardManager.FeePayment(DEFAULT_CONFIG_DIGEST, uint192(DEFAULT_REPORT_LINK_FEE));
+    payments[3] = IDestinationRewardManager.FeePayment(DEFAULT_CONFIG_DIGEST, uint192(DEFAULT_REPORT_LINK_FEE));
+    payments[4] = IDestinationRewardManager.FeePayment(DEFAULT_CONFIG_DIGEST, uint192(DEFAULT_REPORT_LINK_FEE));
+
+    vm.expectEmit();
+
+   bytes32[] memory poolIds = new bytes32[](5);
+    for (uint256 i = 0; i < 5; ++i) {
+      poolIds[i] = payments[i].poolId;
+    }
+
+    emit InsufficientLink(payments);
+
+    processFee(poolIds, payloads, USER, address(native), 0);
+
+    assertEq(getNativeBalance(address(feeManager)), DEFAULT_REPORT_NATIVE_FEE * 5);
+    assertEq(getNativeBalance(USER), DEFAULT_NATIVE_MINT_QUANTITY - DEFAULT_REPORT_NATIVE_FEE * 5);
+    assertEq(getLinkBalance(USER), DEFAULT_LINK_MINT_QUANTITY);
+  }
+
+
+  function test_processPoolIdsPassedMismatched() public {
```

Comment:
@Fletch153 added this test because I saw you added: `error PoolIdMismatch`
this test currently fails.

Reply:
thats what this test does 🙏🏼